src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -8,7 +8,25 @@
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidDataException {
+        if (user.getAge() != null && user.getLogin() != null && user.getPassword() != null) {
+            if (storageDao.get(user.getLogin()) != null) {
+                if (user.getLogin().equals(storageDao.get(user.getLogin()).getLogin())) {
```

Comment:
The check `user.getLogin().equals(storageDao.get(user.getLogin()).getLogin())` is redundant because `storageDao.get(user.getLogin()) != null` already ensures that the login exists. Consider simplifying this logic.